### PR TITLE
fix kfctl corner cases

### PR DESCRIPTION
**Pull request by kunmingg**
* SetApplicationParameter will not throw error if application not found.
  * application might be deleted from config file on purpose
* For same reason we print warnings instead of error when default namespace not found.
* Also update cmd help messages to include local config example.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4380)

### CODE REVIEW

File: bootstrap/pkg/kfapp/kustomize/kustomize.go [hunk 1]
```diff
@@ -204,6 +204,8 @@ func (kustomize *kustomize) Apply(resources kftypesv3.ResourceEnum) error {
 		}
 		return nil
 	}, b)
+	log.Warnf("Default namespace creation skipped")
```
Comment (lluunn) [inline bootstrap/pkg/kfapp/kustomize/kustomize.go]: should we check err != nil and then print warning?

File: bootstrap/pkg/apis/apps/kfconfig/types.go [hunk 2]
```diff
@@ -571,6 +571,8 @@ func (c *KfConfig) SetApplicationParameter(appName string, paramName string, val
 
 			return nil
 		}
+		log.Warnf("Application %v not found", appName)
```
Comment (lluunn) [inline bootstrap/pkg/apis/apps/kfconfig/types.go]: should we do this outside the if at line 577?

File: bootstrap/pkg/kfapp/gcp/gcp.go [hunk 3]
```diff
@@ -1809,12 +1809,17 @@ func (gcp *Gcp) ConfigPodDefault() error {
 		return kfapis.NewKfErrorWithMessage(err, "set K8s clientset error")
 	}
 
+	defaultNamespace := kftypesv3.EmailToDefaultName(gcp.kfDef.Spec.Email)
+	_, err = k8sClient.CoreV1().Namespaces().Get(defaultNamespace, metav1.GetOptions{})
+	if err != nil {
+		log.Warnf("Default namespace %v creation skipped", defaultNamespace)
+		return nil
+	}
```
Comment (gabrielwen) [inline bootstrap/pkg/kfapp/gcp/gcp.go]: why this?  the default namespace is created with the default profile in kustomize.go, right?

Comment (kunmingg) [reply bootstrap/pkg/kfapp/gcp/gcp.go]: Currently default profile is one entry application entry from manifest. If user delete this entry default namespace will not be created.

**Comment by kunmingg** [comment]
If default profile is skipped then default namespace + pod-default will be skipped as well.

**Pull request merged**